jupyter/restore_raft_quorum_with_tablets.ipynb

```diff
@@ -0,0 +1,680 @@
+   "metadata": {},
+   "source": [
+    "## Step 9.\n",
+    "Rolling restart all live nodes, but the recovery leader must be restarted first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "545d32ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "cluster.restart_scylla(nodes=[recovery_coordinator])\n",
+    "cluster.restart_scylla(nodes=[node for node in base_alive_nodes if node != recovery_coordinator])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4691849",
+   "metadata": {},
+   "source": [
+    "## Step 10.\n",
+    "Remove/replace all dead nodes\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "617dc05e",
+   "metadata": {},
+   "source": [
+    "### step 10.1\n",
+    "Remove stopped node in alive dc. it is possible because each rack had 2 nodes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "22807189",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "verification_node.run_nodetool(f\"removenode {stopped_node_host_id} --ignore-dead-nodes {','.join(dead_node_host_ids)}\")\n",
+    "cluster.nodes.remove(host_id_node_map[stopped_node_host_id])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88f56cae",
+   "metadata": {},
+   "source": [
+    "### step 10.2\n",
+    "Replace all nodes in dead dc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "adc25935",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# print(list(dead_node_host_ids_maps.keys())) \n",
+    "# k = list(dead_node_host_ids_maps.keys())[0]\n",
+    "# del dead_node_host_ids_maps[k]\n",
+    "# print(list(dead_node_host_ids_maps.keys())) \n",
+    "\n",
+    "replace_nodes_by_host_id(cluster, dead_node_host_ids_maps, verification_node)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5b04413",
+   "metadata": {},
+   "source": [
+    "#### Step 10.3\n",
+    "Add new nodes to racks\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "597a06c9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dc_idx = cluster.params.region_names.index(dead_dc_region)\n",
+    "for rack in range(2):\n",
+    "    new_node = add_node_to_dc(cluster, dc_idx, rack)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ab0bc430",
+   "metadata": {},
+   "source": [
+    "### 10.4\n",
+    "Add new node instead of remove node in alive dc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9f64b356",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dc_idx = cluster.params.region_names.index(alive_dc_region)\n",
+    "new_node_instead_of_stopped = add_node_to_dc(cluster, dc_idx, rack=stopped_alive_node.rack)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "edba70d7",
+   "metadata": {},
+   "source": [
+    "## Step 11.\n",
+    "Unset recovery_leader on all nodes."
```

Comment:
We need to send the `SIGHUP` signal too (see https://docs.scylladb.com/manual/master/troubleshooting/handling-node-failures.html#manual-recovery-procedure, step 8).